app/__init__.py

```diff
@@ -170,3 +179,10 @@ class InvalidRequest(Exception):
     def __init__(self, message, code=400):
         self.message = message
         self.code = code
+
+
+class ValidationFailed(Exception):
+    def __init__(self, errors, page_count=None, code=400):
+        self.errors = errors
```

Comment:
just a thought, it might be good to define the dict containing all the possible errors in one place. That way, if we add new functions that do different validation checks, there's still one place that contains the list of all values, and each function doesn't need to worry about all the other values that other functions might add. This json is a boundary contract so good to be explicit about it in an easily located place 🙂

```python
self.errors = {
    "content_outside_of_printable_area": [],
    "document_not_a4_size_portrait_orientation": [],
}

for key, value in errors:
    if key in self.errors:
        self.errors[key] = value
```